### PR TITLE
refactor(core): rename Security Groups to Firewalls

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Some people think "Security Groups" is a little too cloud provider specific, so, like we did with "server groups", we are migrating to a more abstract term: "Firewalls".

If this will be confusing to some folks, it can be toggled back to the legacy term of "Security Groups" via a field in the settings (`useClassicFirewallLabels`).

### CODE REVIEW

File: app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ describe('Service: azureServerGroupConfiguration', function() {
       expect(this.command.viewState.securityGroupConfigured).toBeTrue;
     });
 
-    it('finds matcing security groups, sets dirty flag for subsequent time', function() {
```
Comment (jrsquared) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js]: i will miss finding matcing things.

Comment (anotherchrisberry) [reply app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js]: so will karma

File: app/scripts/modules/amazon/src/serverGroup/details/sections/SecurityGroupsDetailsSection.tsx [hunk 2]
```diff
@@ -59,12 +59,12 @@ export class SecurityGroupsDetailsSection extends React.Component<
...
 
     return (
-      <CollapsibleSection heading="Security Groups">
+      <CollapsibleSection heading={FirewallLabels.get('Firewalls')}>
         <ul>
           {sortBy(securityGroups, 'name').map(securityGroup => (
             <li key={securityGroup.name}>
               <UISref
-                to="^.securityGroupDetails"
+                to="^.firewallDetails"
```
Comment (erikmunson) [inline app/scripts/modules/amazon/src/serverGroup/details/sections/SecurityGroupsDetailsSection.tsx]: what's the story behind updating this one but not all the other links to `^.securityGroupDetails`?

File: settings.js [hunk 3]
```diff
@@ -21,6 +21,7 @@ var templatesEnabled = process.env.TEMPLATES_ENABLED === 'true';
 var atlasWebComponentsUrl = process.env.ATLAS_WEB_COMPONENTS_URL;
 var canaryAccount = process.env.CANARY_ACCOUNT || '';
 var canaryFeatureDisabled = process.env.CANARY_FEATURE_ENABLED !== 'true';
+var useClassicFirewallLabels = process.env.USE_CLASSIC_FIREWALL_LABELS !== 'true';
```
Comment (erikmunson) [inline settings.js]: should this be `===` instead of `!==`?

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts [hunk 4]
```diff
@@ -285,6 +286,7 @@ export class ApplicationDataSource implements IDataSourceConfig {
     if (!config.label && this.$filter) {
       this.label = this.$filter('robotToHuman')(config.key);
     }
+    this.label = FirewallLabels.get(this.label);
```
Comment (erikmunson) [inline app/scripts/modules/core/src/application/service/applicationDataSource.ts]: am i inferring correctly that this overrides the label defined in `securityGroup.dataSource.ts`?

File: settings.js (hunk 3 above)
Comment (anotherchrisberry) [reply settings.js]: :cries

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts (hunk 4 above)
Comment (anotherchrisberry) [reply app/scripts/modules/core/src/application/service/applicationDataSource.ts]: You are correctly inferring.

File: app/scripts/modules/amazon/src/serverGroup/details/sections/SecurityGroupsDetailsSection.tsx (hunk 2 above)
Comment (anotherchrisberry) [reply app/scripts/modules/amazon/src/serverGroup/details/sections/SecurityGroupsDetailsSection.tsx]: Thanks for catching - I had started to convert the routes but forgot about it. I've updated the PR to convert all the routes.

**Comment by anotherchrisberry** [comment]
Thanks for catching that, @sbwsg - it was an oversight on my part. There is no good story around changing the labels on recent items, so I am going to leave it as-is right now, and follow up with a separate PR to address this.

**Pull request merged**